packages/fast-check/src/arbitrary/option.ts

```diff
@@ -13,8 +13,8 @@ import { safeHasOwnProperty } from '../utils/globals';
  */
 export interface OptionConstraints<TNil = null> {
   /**
-   * The probability to build a nil value is of `1 / freq`
```

Comment:
The fact that we are changing the default makes me think that it should not be changed into a minor/patch without extra care. What I did in such cases for the previous version of fast-check has been to introduce a `next:boolean` in the constraints object. When true we toggle on the new behavior, otherwise we preserve the old one 

For the record, I may plan a major sooner than expected to be able to drop cjs+esm bundlings.